nipy/labs/viz_tools/maps_3d.py

```diff
@@ -117,14 +117,15 @@ def m2screenshot(mayavi_fig=None, mpl_axes=None, autocrop=True):
     if mpl_axes is not None:
         plt.axes(mpl_axes)
 
-    filename = tempfile.mktemp('.png')
+    fd, filename = tempfile.mkstemp('.png')
```

Comment:
Will this also work on Windows?   I mean - writing to the `filename` while the file is already open with file descriptor `fd`?   And then reading?

Reply:
It also works on Windows. Opening/writing to a file that already has a file-descriptor opened is possible.

You can have a look at this example - 
![image](https://github.com/nipy/nipy/assets/138655107/5e69ea23-eb5d-49fb-917f-da0bcf6acfad)

Here, I'm opening the same file twice. In the context manager, when I open the file (`f`), it gets the file-descriptor value of 4 assigned to it. And I while writing to the file, the first `fd: 3` is still opened. So, it's not a problem.